features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/APIDefinition/APIDefinition.jsx

```diff
@@ -0,0 +1,276 @@
+            readOnly: true,
+            smoothScrolling: true,
+            wordWrap: 'on',
+            handleMouseWheel: true,
+            scrollbar: {
+                handleMouseWheel: true
+            }
+        };
+        if (this.state.notFound) {
+            return <ResourceNotFound message={this.props.resourceNotFountMessage}/>;
+        }
+        if (!swagger) {
+            <Progress />
+        }
+
+        return (
+            <div className={classes.root}>
+                <div className={classes.titleWrapper}>
+                    <Typography variant='h4' align='left' className={classes.mainTitle}>
+                        API Definition
+                    </Typography>
+                    <Button size='small' className={classes.button} onClick={this.openEditor}>
+                        <EditRounded className={classes.buttonIcon}/>
+                        Edit
+                    </Button>
+                    <Dropzone
+                        multiple={false}
+                        className={classes.dropzone}
+                        accept={['application/json', 'application/x-yaml']}
+                        onDrop={(dropFile) => {
+                            this.onDrop(dropFile);
+                        }}>
+                        <Button size='small' className={classes.button}>
+                            <CloudUploadRounded className={classes.buttonIcon}/>
+                            Import API Definition
+                        </Button>
+                    </Dropzone>
+                </div>
+                <div>
+                    <MonacoEditor
+                        width='100%'
+                        height='calc(100vh - 51px)'
+                        theme='vs-dark'
+                        value={swagger}
+                        options={editorOptions}
+                    />
+                </div>
+                <Dialog fullScreen open={this.state.openEditor} onClose={this.closeEditor}
+                        TransitionComponent={this.Transition}>
+                    <Paper square className={classes.popupHeader}>
+                        <IconButton color='inherit' onClick={this.closeEditor} aria-label='Close'>
+                            <Icon>close</Icon>
+                        </IconButton>
+
+                        <Button className={classes.button} variant='contained' color='primary'
+                                onClick={this.updateSwaggerContent}>
+                            <FormattedMessage
+                                id='documents.swagger.editor.update.content'
+                                defaultMessage='Update Content'
+                            />
+                        </Button>
+                    </Paper>
+                    <SwaggerEditorDrawer />
+                </Dialog>
+            </div>
+        );
+    }
+}
+APIDefinition.propTypes = {
+    classes: PropTypes.shape({}).isRequired,
+    state: PropTypes.shape({}).isRequired,
```

Comment:
Are we accepting a prop named `state` ??

Reply:
fixed https://github.com/wso2/carbon-apimgt/pull/6572/commits/547554b9a9c260c3115ce9cd4e74ad8008c3d85a